src/main/java/net/blay09/mods/waystones/block/WaystoneBlock.java

```diff
@@ -145,10 +172,17 @@ public void onReplaced(BlockState state, World world, BlockPos pos, BlockState n
         super.onReplaced(state, world, pos, newState, isMoving);
 
         // Also destroy the connect upper or lower waystone block
+        final IFluidState fluidState = world.getFluidState(pos);
         if (world.getBlockState(pos.up()).getBlock() == this) {
-            world.removeBlock(pos.up(), false);
+            if (fluidState.getFluid() == Fluids.WATER) {
+                world.setBlockState(pos, fluidState.getBlockState(), 35);
+                world.removeBlock(pos.up(), false);
```

Comment:
Instead of duplicating the line, removeBlock can be moved outside of the if and make the else unnecessary

Reply:
Thanks! Thats a good point